sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc

```diff
@@ -1167,6 +1192,74 @@ Exceptions:
   `property::graph::updatable` was not set when the executable graph was created.
 * Throws with error code `invalid` if any node in `nodes` is not part of the
   graph.
+
+|
+[source, c++]
+----
+void
+update(const command_graph<graph_state::modifiable>& graph);
+----
+
+|Updates all of the nodes in the executable graph with parameters from a
+topologically identical modifiable graph. A topologically identical graph is one
+with the same structure of nodes and edges. Additionally, all nodes and edges
+must be added in the same order, and all Graph-related API calls and recorded
+queue submissions must occur in the same order.
+
+Equivalent nodes in topologically identical graphs each have the
+same command. There is the additional limitation that to update an executable
```

Comment:
What does "same command" mean?  Consider a kernel that is a functor object:

```
class MyKernel {
  // ...
};

void dothing(void *ptr, sycl::queue q) {
  MyKernel k(ptr);
  q.submit([&](sycl::handler& cgh) {
    cgh.parallel_for({N}, k);
  });
}
```

Let's say I record graph _G1_ by calling `dothing` execute that graph, record a second graph _G2_ by calling `dothing`, and then try to update _G1_ using _G2_ as a source graph.  Both graphs have one node.  Do the two nodes satisfy the requirement of "same command"?  Note that each call to `dothing` constructs a new `k` kernel object, so the two nodes execute different kernel objects.

Reply:
I think they would satisfy this requirement. We could check the internal kernel objects which should be the same in this case, though that may be problematic if we allow kernel command update in future. So I think the onus still has to be on the user to ensure that they are the same in this case. We could provide more detail about what the same command means though, something like "the kernel functions must have identical types, whether that be a lambda function, function object or otherwise".

Reply:
> We could provide more detail about what the same command means though, something like "the kernel functions must have identical types, whether that be a lambda function, function object or otherwise".

Yes, I think that would be more precise.